### ISSUE TITLE
Babel sourcemaps  support

### ISSUE THREAD

**Issue by marthinus-engelbrecht** [cucumber/cucumber-js#567]
Hi,

Thanks for this. Cucumber + javascript = awesome. 
 
I'm using babel as a compiler. Is there a way that I can tell cucumberjs to use the sourcemaps?

**Comment by charlierudolph** [comment]
Use them in what way? In my experience with other compilers, they hook automatically to error stacks. If cucumber-js is loading the babel files, I expect the sourcemaps to be used.

**Comment by marthinus-engelbrecht** [comment]
I must confess, I really don't know much about how source maps work. All I know is when I include them in the directory I'm serving the browser automagically figures it out by looking at the source maps comment at the bottom of the generated file. It seems however that node doesn't figure it out just by it self. I had to use   [node-source-map-support](https://github.com/evanw/node-source-map-support) to get node to understand the source-maps of a little test program I wrote. 

Looking at it more carefully though it seems that cucumber js is figuring out where to find the source maps but it's reporting the incorrect line for the step definition. See below: 

```
Scenario:  - test/functional/generateNodeModule/byAnsweringQuestions.feature:5
   Step: When the generator is run without options - test/functional/generateNodeModule/byAnsweringQuestions.feature:6
   Step Definition: test/functional/generateNodeModule/steps.js:217
```

The correct line is 77.

**Comment by charlierudolph** [comment]
Hmm, can you please provide a sample project setup so I can reproduce?

**Comment by marthinus-engelbrecht** [comment]
Will put something together.

**Comment by marthinus-engelbrecht** [comment]
Ok, clone [this](https://github.com/marthinus-engelbrecht/cucumber_no_source_map). You can read the instructions on the README. I actually don't expect it to do anything else because I'm not telling it where to find or how to produce source maps.

**Comment by charlierudolph** [comment]
I get the following output when running on node 6
```
> cucumber_no_source_map@1.0.0 test /Users/charlesrudolph/github/cucumber_no_source_map
> cucumber-js --format summary --compiler js:babel-register tests/

when block
and block
Failures:

1) Scenario:  - tests/test.feature:4
   Step: Then It should give the correct step definition line number - tests/test.feature:7
   Step Definition: tests/steps.js:15
   Message:
     Error: Error on the it block
         at World._callee$ (steps.js:12:15)
         at tryCatch (/Users/charlesrudolph/github/cucumber_no_source_map/node_modules/babel-regenerator-runtime/runtime.js:61:40)
         at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/charlesrudolph/github/cucumber_no_source_map/node_modules/babel-regenerator-runtime/runtime.js:329:22)
         at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Users/charlesrudolph/github/cucumber_no_source_map/node_modules/babel-regenerator-runtime/runtime.js:94:21)
         at step (/Users/charlesrudolph/github/cucumber_no_source_map/tests/steps.js:3:191)
         at /Users/charlesrudolph/github/cucumber_no_source_map/tests/steps.js:3:451
```
The line appears correct to me.

**Comment by marthinus-engelbrecht** [comment]
There is no line 15 in steps.js

**Comment by bdo** [comment]
@marthinus-engelbrecht `steps.js:12:15` indicates line 12 column 15, so it seems to be working properly (at least on node 6)

**Comment by charlierudolph** [comment]
The error stack works (`Message:` line) properly but when we try to determine the location of the step (`Step Definition:` line) the source map doesn't appear to be taken into account

**Comment by bdo** [comment]
Oh I see. My bad.

**Comment by charlierudolph** [comment]
Fixed in 1d02c6af66763189ef8ffc18880245ce39a56172

**Comment by marthinus-engelbrecht** [comment]
Awesome 👍
